diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0c9767..33050c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -331,10 +331,6 @@ function(check_flag lang flag)
     string(REGEX REPLACE "[^A-Za-z0-9]" "_" flag_var "${flag}")
     if(NOT DEFINED ${lang}_FLAG${flag_var})
         execute_process(COMMAND ${CMAKE_COMMAND} -E echo_append "-- Checking ${lang} flag support for: [${flag}] - ")
-        if(CMAKE_C_COMPILER_ID MATCHES "Clang")
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=unused-command-line-argument")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=unused-command-line-argument")
-        endif()
         if("${lang}" STREQUAL "CXX")
             check_cxx_compiler_flag("${flag}" "${lang}_FLAG${flag_var}")
         else()
@@ -520,8 +516,7 @@ set_property(DIRECTORY .
         SAFECLIB_STR_NULL_SLACK=1
         $<$<BOOL:${HAVE_STRNLEN_S}>:HAVE_STRNLEN_S=1>
         $<$<BOOL:${HAVE_STRNCPY_S}>:HAVE_STRNCPY_S=1>
-        $<$<BOOL:${HAVE_STRCPY_S}>:HAVE_STRCPY_S=1>
-        $<$<BOOL:${WIN32}>:_WIN32_WINNT=0x0601>)
+        $<$<BOOL:${HAVE_STRCPY_S}>:HAVE_STRCPY_S=1>)
 if(NOT HAVE_STRCPY_S OR NOT HAVE_STRNCPY_S OR NOT HAVE_STRNLEN_S)
     add_library(safestringlib OBJECT
         third_party/safestringlib/safeclib_private.h
