diff --git a/CMakeLists.txt b/CMakeLists.txt
index da1917e..06d7203 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,13 +47,12 @@ if(SPNG_STATIC)
     list(APPEND spng_TARGETS spng_static)
 endif()

-find_package(ZLIB REQUIRED)
 foreach(spng_TARGET ${spng_TARGETS})
     target_include_directories(${spng_TARGET} PUBLIC
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/spng>
       $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
     )
-    target_link_libraries(${spng_TARGET} PRIVATE ZLIB::ZLIB)
+    target_link_libraries(${spng_TARGET} PRIVATE zlib-ng-static)
     target_link_libraries(${spng_TARGET} PRIVATE ${MATH_LIBRARY})
 endforeach()

