From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
Date: Thu, 25 Jun 2026 00:00:00 +0800
Subject: [PATCH] CMake: Do not link librt on Android

---
 absl/base/CMakeLists.txt | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/absl/base/CMakeLists.txt b/absl/base/CMakeLists.txt
--- a/absl/base/CMakeLists.txt
+++ b/absl/base/CMakeLists.txt
@@ -14,7 +14,12 @@
 # limitations under the License.
 #
 
-find_library(LIBRT rt)
+if(ANDROID OR "${TARGET_OS}" STREQUAL "ANDROID")
+  set(ABSL_LIBRT_LINKOPTS)
+else()
+  find_library(LIBRT rt)
+  set(ABSL_LIBRT_LINKOPTS $<$<BOOL:${LIBRT}>:-lrt>)
+endif()
 
 # Internal-only target, do not depend on directly.
 absl_cc_library(
@@ -273,7 +277,7 @@ absl_cc_library(
     ${ABSL_DEFAULT_COPTS}
   LINKOPTS
     ${ABSL_DEFAULT_LINKOPTS}
-    $<$<BOOL:${LIBRT}>:-lrt>
+    ${ABSL_LIBRT_LINKOPTS}
     $<$<BOOL:${MINGW}>:-ladvapi32>
   DEPS
     absl::atomic_hook
